src/plugins/lua/weechat-lua.c

```diff
@@ -251,19 +251,27 @@ weechat_lua_output_flush ()
 }
 
 /*
+ * NOTE: Taken from `luaB_print`
  * Redirection for stdout and stderr.
  */
 
 int
-weechat_lua_output (lua_State *L)
-{
-    const char *msg, *ptr_msg, *ptr_newline;
-
-    if (lua_gettop (L) < 1)
-        return 0;
-
-    msg = lua_tostring (L, -1);
-    ptr_msg = msg;
+weechat_lua_output (lua_State *L) {
```

Comment:
Please reformat and use code style described in the doc here: https://weechat.org/files/doc/devel/weechat_dev.en.html#coding_c_style